Add Table Policy documentation (#72)
crate/cloud-docs: docs/reference/overview.md

  potential delays.
:::

(overview-table-policies)=
### Table Policies

Table policies allow to automate maintenance operations for 
**partitioned tables**. Automated actions can be set up that are be 
executed daily based on pre-configure ruleset.

![Table policy list](../_assets/img/cluster-table-policy.png)
:::

Table policy overview can be found in the left-hand navigation menu under 
"Table Policies". From the list of policies, you can create, delete, edit, 
or (de)activate them. 

Log of executed policies can be found in the "Logs" tab. 

![Table policy list](../_assets/img/cluster-table-policy-logs.png)
:::

> New policy can be created with "Add New Policy" button.

Reviewer: ```suggestion
A new policy can be created with "Add New Policy" button.
```

![Table policy list](../_assets/img/cluster-table-policy-create.png)
:::

After you give the policy a name and choose the tables/schemas which should
be affected, you need to choose the time column. The time column will be used 
with the condition to select which data will be affected. This column should 
be preset in all the tables/schemas that the policy is using.

:::{note}
The "Time Column" must be of type `TIMESTAMP`.
:::

Next, a condition is used to determine affected partitions. The system is
time based. A partition is eligible for action if the value in the 
partitioned column is smaller (`<`), or smaller or equal (`<=`) than the current
date minus N days, months, or years.

Following actions are supported:
* Delete: Deletes eligible partitions along with their data.
* Set replicas: Changes the replication factor of eligible partitions.
* Force merge: Merges segments on eligible partitions to ensure a specified
  number.

> After filling out the info, you can see the affected schemas/tables and
> the number of affected partitions at the next iteration of the action.

Reviewer: ```suggestion
After filling out the info, you can see the affected schemas/tables and
the number of affected partitions if the policy gets executed at this very moment.
```

or (de)activate them. 

Log of executed policies can be found in the "Logs" tab. 

![Table policy list](../_assets/img/cluster-table-policy-logs.png)
:::

A new policy can be created with "Add New Policy" button.

![Table policy list](../_assets/img/cluster-table-policy-create.png)
:::

After naming the policy and selecting the tables/schemas to be impacted, you
must specify the time column. This column, which should be a timestamp used for
partitioning, will determine the data affected by the policy. It is important
that this time column is consistently present across all targeted
tables/schemas. While you can apply the policy to also tables without the
specificed time column, it will not get executed for those. If your tables have
different timestamp columns, consider setting up separate policies for each to
ensure accuracy.

:::{note}
The "Time Column" must be of type `TIMESTAMP`.
:::

Next, a condition is used to determine affected partitions. The system is
time based. A partition is eligible for action if the value in the 
partitioned column is smaller (`<`), or smaller or equal (`<=`) than the current

> date minus N days, months, or years.

Reviewer: ```suggestion
date minus `n` days, months, or years.
```

Following actions are supported:

> * Delete: Deletes eligible partitions along with their data.
> * Set replicas: Changes the replication factor of eligible partitions.
> * Force merge: Merges segments on eligible partitions to ensure a specified

Reviewer: ```suggestion
* **Delete:** Deletes eligible partitions along with their data.
* **Set replicas:** Changes the replication factor of eligible partitions.
* **Force merge:** Merges segments on eligible partitions to ensure a specified
```

maybe make it bold?

that this time column is consistently present across all targeted
tables/schemas. While you can apply the policy to also tables without the
specificed time column, it will not get executed for those. If your tables have
different timestamp columns, consider setting up separate policies for each to
ensure accuracy.

:::{note}
The "Time Column" must be of type `TIMESTAMP`.
:::

Next, a condition is used to determine affected partitions. The system is
time based. A partition is eligible for action if the value in the 
partitioned column is smaller (`<`), or smaller or equal (`<=`) than the current
date minus `n` days, months, or years.

(overview-table-policies-actions)=
#### Actions

Following actions are supported:
* **Delete:** Deletes eligible partitions along with their data.
* **Set replicas:** Changes the replication factor of eligible partitions.
* **Force merge:** Merges segments on eligible partitions to ensure a specified
  number.

After filling out the info, you can see the affected schemas/tables and
the number of affected partitions if the policy gets executed at this very moment.

(overview-table-policies-usage)=

> #### Usage
> 
> Some examples of the Table Policy use cases are: 
> * Removal of data older than n days
> * Setting replicas to 0 to save replica space 
> 
> Let's assume the following table schema:
> 
> :::{code} sql
> CREATE TABLE data_table (
>    ts TIMESTAMP,
>    ts_day GENERATED ALWAYS AS date_trunc('day',ts),
>    val DOUBLE
> ) PARTITIONED BY (ts_day);
> :::
> 
> In this case, the configuration for the two use cases would be following:
> 
> **Data removal:**
> * **Time Columnd:** `ts_day`
> * **Condition:** `< 60 days`
> * **Actions:** `Delete eligible partition(s)`
> 
> **Saving replica space:**
> * **Time Columnd:** `ts_day`
> * **Condition:** `< 30 days`
> * **Actions:** `Set 0 replicas.`
> 

Reviewer: ```suggestion
#### Examples

Consider a scenario where you have a table and wish to optimize space on your
cluster. For older data, which might already be snapshoted, it may be sufficient
for it to exist just once in the cluster without replication. In such cases,
high availability is not a priority, and you plan to retain the data for only
60 days.

Assume the following table schema:

:::{code} sql
CREATE TABLE data_table (
   ts TIMESTAMP,
   ts_day GENERATED ALWAYS AS date_trunc('day',ts),
   val DOUBLE
) PARTITIONED BY (ts_day);
:::

For the outlined scenario, the policies would be as follows:

**Policy 1 - Saving replica space:**
* **Time Column:** `ts_day`
* **Condition:** `older than 30 days`
* **Actions:** `Set replicas to 0.`

**Policy 2 - Data removal:**
* **Time Column:** `ts_day`
* **Condition:** `older than 60 days`
* **Actions:** `Delete eligible partition(s)`
```

like this maybe, similiar to job scheduler?
